src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointPair.java

```diff
@@ -0,0 +1,476 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.util.SequenceUtil;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.BreakpointPair.EvidenceFlankSide.*;
+
+/**
+ * This class represents inferred a pair of breakpoints of a simple SV event
+ * (a simple rearrangement between two genomic locations)
+ * that is suggested by the input {@link ChimericAlignment},
+ * and complications in pinning down the breakpoints's location to exact base pair resolution.
+ */
+@DefaultSerializer(BreakpointPair.Serializer.class)
+class BreakpointPair {
+
+    static final String EMPTY_STRING = "";
+    private static final int MAX_LOWER_CN = 10; // in case of CNV events involving tandem duplications, the maximum value of the lower copy number the algorithm will try
+
+    final SimpleInterval leftJustifiedLeftBreakpoint;
+    final SimpleInterval leftJustifiedRightBreakpoint;
+
+    final EvidenceFlankSide evidenceFlankSide;
+    final BreakpointComplication complication;
+
+
+    /**
+     * Represents the strand of evidence that was used in computing this breakpoint pair.
+     */
+    enum EvidenceFlankSide {
+        BOTH, THREE_PRIME, FIVE_PRIME
+    }
+
+    /**
+     * A helper struct for annotation breakpoints represented by this BreakpointPair.
+     * To be updated as more types of complications can be mentally processed and
+     * handled by {@link #resolveComplications(ChimericAlignment)}.
+     */
+    @DefaultSerializer(BreakpointComplication.Serializer.class)
+    static final class BreakpointComplication {
```

Comment:
I was trying to be future-proof (maybe over-shooting), because I believe we might need to update this in more complicated calling, like in this case it is already incorporating the tandem duplication (albeit vey naively) annotations that must be included to tell downstream steps that it is not as simple as a simple insertion or deletion.